File: ReactNative/appreveal/android/src/main/kotlin/com/appreveal/elements/ElementInfo.kt
package com.appreveal.elements

/**
 * Type of UI element. Values match iOS ElementType rawValue exactly.
 */
enum class ElementType(val value: String) {
    BUTTON("button"),
    TEXT_FIELD("textField"),
    LABEL("label"),
    IMAGE("image"),
    TOGGLE("toggle"),
    SLIDER("slider"),
    STEPPER("stepper"),
    PICKER("picker"),
    SCROLL_VIEW("scrollView"),
    TABLE_VIEW("tableView"),
    COLLECTION_VIEW("collectionView"),
    CELL("cell"),
    NAVIGATION_BAR("navigationBar"),
    TAB_BAR("tabBar"),
    OTHER("other")
}

/**
 * Describes a visible interactive element on screen.
 * Matches iOS ElementInfo exactly.
 */
data class ElementInfo(
    val id: String,
    val type: ElementType,
    val label: String?,
    val value: String?,
    val enabled: Boolean,
    val visible: Boolean,
    val tappable: Boolean,
    val frame: ElementFrame,
    val containerId: String?,
    val actions: List<String>,
    /** How the id was derived: "explicit", "text", "semantics", "derived" */
    val idSource: String
) {
    data class ElementFrame(
        val x: Double,
        val y: Double,
        val width: Double,
        val height: Double
    )
}

File: iOS/Sources/AppReveal/Elements/ElementInfo.swift
// Element model for UI element inventory

import Foundation

#if DEBUG

/// Type of UI element.
public enum ElementType: String, Codable {
    case button
    case textField
    case label
    case image
    case toggle
    case slider
    case stepper
    case picker
    case scrollView
    case tableView
    case collectionView
    case cell
    case navigationBar
    case tabBar
    case other
}

/// Describes a visible interactive element on screen.
public struct ElementInfo: Codable {
    public let id: String
    public let type: ElementType
    public let label: String?
    public let value: String?
    public let enabled: Bool
    public let visible: Bool
    public let tappable: Bool
    public let frame: ElementFrame
    public let containerId: String?
    public let actions: [String]
    /// How the id was derived: "explicit", "text", "semantics", "tooltip", "derived"
    public let idSource: String

    public struct ElementFrame: Codable {
        public let x: Double
        public let y: Double
        public let width: Double
        public let height: Double
    }
}

#endif
